tools/testbench/testbench.c

```diff
@@ -306,6 +306,7 @@ int main(int argc, char **argv)
 
 	while (frcd->fs.reached_eof == 0)
 		pipeline_schedule_copy(p, 0);
+		// should we add pipeline_schedule_copy to the second pipeline?
```

Comment:
After knowing that pipeline_schedule_copy() is called for not only one pipeline (as explained in pipeline.c), then I start to doubt this code: here we only put pipeline_schdule_copy() for the first pipeline, should we also add the second one?

Reply:
HOORAY. It works after I added the following codes:

struct pipeline *p2;
pcm_dev = ipc_get_comp_by_id(sof.ipc, 7); // SSP2.OUT
p2 = pcm_dev->cd->pipeline;

while (frcd->fs.reached_eof == 0) {
        pipeline_schedule_copy(p, 0);
        pipeline_schedule_copy(p2, 0);
}

And I have checked the output waveform and they are as expected.
However these codes are fishy, we need to embellish them.